### DISCUSSION TITLE
How to use Mirth for basic file transfer?

### DISCUSSION THREAD

**Discussion by ghost** [nextgenhealthcare/connect#4526] [Q&A]
Hello, 

I wanted to use Mirth for a basic file transfer - I have a bunch of files in one folder and want to transfer them to another one (with some checks happening along the way). I thought of doing this with a File Reader/Writer but read on the Mirth forum that it's probably easier to just do the transfer via JScript if I don't want to change the file. 

How would one do that? :) I'm pretty new to Mirth and JS so I don't really know how to. 

Thanks!

**Comment by tonygermano** [comment]
The details are important here. What type of file? Are they being moved within the same file system or to another one? How big are the files? What kind of checks?

It is generally more performant to use java methods via javascript or even another tool outside or mirth for strictly moving entire files from one place to another. The goal is usually to avoid reading the entire file into memory and writing it back to the database as a message, as a File Reader would do.

However, if you need to process the file to do your checks (whether you make changes or not is irrelevant) before writing it back out, then a File Reader makes sense, because you want to process the file as a message.

**Reply by pacmano1** [reply]
I'd add that without you sharing the network topology, the question is nearly impossible to answer.
